WooCommerce/Classes/Yosemite/AuthenticatedState.swift

```diff
@@ -137,7 +139,11 @@ class AuthenticatedState: StoresManagerState {
         trackEventRequestNotificationHandler = TrackEventRequestNotificationHandler()
 
         startListeningToNotifications()
-        checkApplicationPasswordExperimentFeatureFlag()
+        observeExperimentFeatureSettings()
+
+        DispatchQueue.main.async {
+            self.checkApplicationPasswordExperimentFeatureState()
+        }
```

Comment:
Nit: Should this be simplified: 
```suggestion
        Task { @MainActor in
            await checkApplicationPasswordExperimentFeatureState()
        }
```

Then make `checkApplicationPasswordExperimentFeatureState` async for better testability if we need to test it.